images/win/scripts/Installers/Validate-PostgreSQL.ps1

```diff
@@ -0,0 +1,42 @@
+function Stop-PostgreSQL {
+    param(
+        [String]$PostgresPath
+    )
+    $pgdata=Join-Path $PostgresPath "data"
+    Start-Process -FilePath pg_ctl -ArgumentList ("-D", "$pgdata", "stop")
+    Write-Host "PostgreSQL has been successfully started."
+}
+function Validate-PostgreSQL {
+    $pgready = Start-Process -FilePath pg_isready -Wait -PassThru
+    $exitCode = $pgready.ExitCode
+    if ($exitCode -eq 0)
+    {
+        Write-Host -Object "PostgreSQL has been successfully installed."
+    }
+    else
+    {
+        Write-Host -Object "PostgreSQL is not ready. Exitcode: $exitCode"
+        exit $exitCode
+    }
+}
+
+$paths=(Get-CimInstance Win32_Service -Filter "Name LIKE 'postgresql-%'").PathName
+$pgservice=(Get-CimInstance Win32_Service -Filter "Name LIKE 'postgresql-%'").Name
+$pgbin=$paths.split('"')[1].replace("\pg_ctl.exe", "")
+$env:Path +=";$pgbin"
+$pgroot=$paths.split('"')[1].replace("\bin\pg_ctl.exe", "")
+$psqlVersion=pg_config --version | Out-String
+Validate-PostgreSQL -PostgresPath $pgroot
```

Comment:
Function does't have parameters.

Reply:
Yes, the function does not have parameters, since it was created just to check that postgres service is ready to accept connections. Is it bad practice to use function without parameters?

Reply:
I mean about `-PostgresPath $pgroot` passing parameters to the function without parameters.

Reply:
Ah, clear, my fault. Thank you for the explanation.